src/config.ts

```diff
@@ -485,6 +494,35 @@ export class KubeConfig implements SecurityAuthentication {
         const config: Configuration = createConfiguration({
             baseServer: baseServerConfig,
             authMethods: authConfig,
+            // This HTTP API implementation is a copy of ./gen/http/isomorphic-fetch.ts,
+            // extended with the timeout pass-through code.
+            httpApi: {
+                send(request: RequestContext): Observable<ResponseContext> {
+                    const method = request.getHttpMethod().toString();
+                    const body = request.getBody();
+
+                    const resultPromise = fetch(request.getUrl(), {
+                        method: method,
+                        body: body as any,
+                        headers: request.getHeaders(),
+                        agent: request.getAgent(),
+                        timeout: options?.timeoutMs,
+                    }).then((resp: any) => {
+                        const headers: { [name: string]: string } = {};
+                        resp.headers.forEach((value: string, name: string) => {
+                            headers[name] = value;
+                        });
+
+                        const body = {
+                            text: () => resp.text(),
+                            binary: () => resp.buffer(),
+                        };
+                        return new ResponseContext(resp.status, headers, body);
+                    });
```

Comment:
Not 100% sure but do we need a `catch` here as well or will this be catched by the client code?